### ISSUE TITLE
feat(tooltip): bindToAriaLabel input in matTooltip directive

### ISSUE THREAD

**Issue by LuisTovar0** [angular/components#29797]
### Feature Description

Consider an element where the `matTooltip` directive is applied. The value of the `matTooltip` input [is bound to](https://v18.material.angular.io/components/tooltip/overview#accessibility) that element's `aria-describedby` attribute, which makes perfect sense.

However, there are circumstances where it seems to make more sense to bind the `matTooltip` input's value to the element's `aria-label` instead, so I think there should be an input flag that switches this binding, maybe called `bindToAriaLabel`.

### Use Case

A common use case for tooltips is to label other elements, e.g. icon buttons:

```html
<button mat-icon-button [matTooltip]="searchButtonLabel | piperPipe" [attr.aria-label]="searchButtonLabel | piperPipe">
    <mat-icon fontIcon="search"/>
</button>
```

This kind of repetition could be avoided with this `matTooltip` input:

```html
<button mat-icon-button [matTooltip]="searchButtonLabel | piperPipe" matTooltipBindToAriaLabel> ...
```

**Comment by crisbeto** [comment]
Closing since this can already be achieved by binding to the `aria-label` and it's likely not common enough to warrant a separate public API.
